tests/data/wpdb.php

```diff
@@ -14,8 +14,19 @@
 assertType('array<int, mixed>|void|null', wpdb::get_row(null, 'ARRAY_N'));
 
 // wpdb::get_results()
-assertType('stdClass|null', wpdb::get_results());
-assertType('stdClass|null', wpdb::get_results(null, 'OBJECT'));
-assertType('array<string, stdClass>|null', wpdb::get_results(null, 'OBJECT_K'));
-assertType('array<string, mixed>|null', wpdb::get_results(null, 'ARRAY_A'));
-assertType('array<int, mixed>|null', wpdb::get_results(null, 'ARRAY_N'));
+    /*
+     *                       Any of ARRAY_A | ARRAY_N | OBJECT | OBJECT_K constants.
+	 *                       ARRAY_A | ARRAY_N | OBJECT: return an array of rows indexed from 0 by SQL result row number.
+     *
+     *                       ARRAY_A: Each row is an associative array (column => value, ...),
+     *                       ARRAY_N: a numerically indexed array (0 => value, ...),
+	 *                       OBJECT: an object ( ->column = value ), respectively.
+     *
+     *                       With OBJECT_K: return an associative array of row objects keyed by the value
+	 *                       of each row's first column's value.
+     */
+assertType('array<int, array<string, mixed>>|null', wpdb::get_results(null, 'ARRAY_A'));
+assertType('array<int, array<int, mixed>>|null', wpdb::get_results(null, 'ARRAY_N'));
+assertType('array<int, stdClass>|null', wpdb::get_results());
+assertType('array<int, stdClass>|null', wpdb::get_results(null, 'OBJECT'));
+assertType('array<array-key, stdClass>|null', wpdb::get_results(null, 'OBJECT_K'));
```

Comment:
```suggestion
assertType('array<stdClass>|null', wpdb::get_results(null, 'OBJECT_K'));
```
`array-key` is the default, it works already, you can leave it away. array-key should be `(int|string)` AFAIK because PHP has this semi-magic behaviour where e.g. `['1' => 'foo', 'bar' => 'baz']` ends up being `[1 => 'foo', 'bar' => 'baz']`, notice the numeric-string key becoming an int.

Reply:
Yes, PHP is not your choice of language when handling something important.

As for removing `array-key,`: you are talking to Explicit Viktor!
`array<something>` looks like a list of something without key type being important.
Please always be explicit.

Reply:
I'll leave it in the function map, but we have to get rid of it here, since PHPStan just cannot make use of it for testing the type, it falls to array<something>, we'll see in the new PR I try to make now